diff --git a/src/device.rs b/src/device.rs
index 9813c50..d79bdd9 100644
--- a/src/device.rs
+++ b/src/device.rs
@@ -372,7 +372,7 @@ pub trait Device: Authenticate + GetPasswordSafe + GenerateOtp {
     /// #     Ok(())
     /// # }
     /// ```
-    fn get_major_firmware_version(&self) -> i32 {
+    fn get_major_firmware_version(&self) -> u8 {
         unsafe { nitrokey_sys::NK_get_major_firmware_version() }
     }
 
@@ -393,7 +393,7 @@ pub trait Device: Authenticate + GetPasswordSafe + GenerateOtp {
     /// );
     /// #     Ok(())
     /// # }
-    fn get_minor_firmware_version(&self) -> i32 {
+    fn get_minor_firmware_version(&self) -> u8 {
         unsafe { nitrokey_sys::NK_get_minor_firmware_version() }
     }
 
